### PR TITLE
address clippy and rust lints

### PR DESCRIPTION
**Pull request by oli-obk**

### CODE REVIEW

File: src/priv_io.rs [hunk 1]
```diff
@@ -1,5 +1,4 @@
 use byteorder::ReadBytesExt;
-use net2::TcpStreamExt;
```
Comment (sfackler) [inline src/priv_io.rs]: This needs to stay to support pre-1.9.0 users.

File: src/lib.rs [hunk 2]
```diff
@@ -1,4 +1,4 @@
-//! A pure-Rust frontend for the popular PostgreSQL database.
+//! A pure-Rust frontend for the popular `PostgreSQL` database.
```
Comment (sfackler) [inline src/lib.rs]: Why?

Comment (oli-obk) [inline src/lib.rs]: The `doc_markdown` lint triggers on camel case words outside ticks

Comment (sfackler) [inline src/lib.rs]: That does not seem like a particularly good reason to put PostgreSQL in ticks :P

**Comment by oli-obk** [comment]
addressed comments

**Comment by sfackler** [comment]
Thanks!

**Pull request merged**